src/web/page.rs

```diff
@@ -119,42 +111,145 @@ impl<T: ToJson> Page<T> {
     }
 }
 
-impl<T: ToJson> ToJson for Page<T> {
-    fn to_json(&self) -> Json {
-        let mut tree = BTreeMap::new();
+impl<T: Serialize> Serialize for Page<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Page", 10)?;
 
         if let Some(ref title) = self.title {
-            tree.insert("title".to_owned(), title.to_json());
+            state.serialize_field("title", title)?;
         }
```

Comment:
Same bug as above with `serialize_struct` and a hard-coded value.

Reply:
Hard coded value?

Reply:
The `10` argument for `serialize_struct`.